lib/filter/filter-expr-grammar.ym

```diff
@@ -103,12 +103,13 @@ filter_simple_expr
 	| KW_NETMASK '(' string ')'    { $$ = filter_netmask_new($3); free($3); }
   | KW_NETMASK6 '(' string ')'   {
   #if ENABLE_IPV6
-                                    $$ = filter_netmask6_new($3);
+    if (!$3) YYERROR;
+    $$ = filter_netmask6_new($3);
   #else
-                                    YYERROR;
+    YYERROR;
   #endif
-                                    free($3);
-                                  }
+    free($3);
```

Comment:
Is the `free()` call in the good place? I think we should put it after `filter_netmask6_new()` and before the 108th line.